src/include/buffer/buffer_pool_manager.h

```diff
@@ -17,20 +17,117 @@
 #include <unordered_map>
 
 #include "buffer/clock_replacer.h"
+#include "gtest/gtest.h"
 #include "recovery/log_manager.h"
 #include "storage/disk/disk_manager.h"
 #include "storage/page/page.h"
 
 namespace bustub {
 
+enum FuncType { FetchPage, UnpinPage, FlushPage, NewPage, DeletePage, FlushAllPages };
```

Comment:
Switch to `enum class`.
Also, `FuncType` is a generic name but these functions are specific to the `BufferPoolManager`. You should rename to `BufferPoolManagerMethodType`.